src/endpoint/s3/s3_rest.js

```diff
@@ -334,7 +334,7 @@ function _get_arn_from_req_path(req) {
 // we will reintreduce it together with bucket site support.
 function parse_bucket_and_key(req) {
     const { url, headers, virtual_hosts } = req;
-    const host = headers.host.split(':')[0]; // cutting off port
+    const host = headers?.host?.split(':')[0]; // cutting off port
 
```

Comment:
@naveenpaul1 do we need to throw here an error about this missing host header?
how AWS react to this missing host header?

Reply:
@romayalon I am not sure how to send the request to AWS s3 without host header, In doc I could see host header is required property for AWS S3. [doc ](https://docs.aws.amazon.com/AmazonS3/latest/API/RESTCommonRequestHeaders.html)
For testing this fix I have to remove the host header inside the code.

Reply:
@romayalon I sent `https` request to AWS S3 url  `{bucket-name}.s3.amazonaws.com` without any header and got `AccessDenied` error. Do we need to return the same error if header is missing?

Reply:
@naveenpaul1 if you attach the host header, do you have access to the bucket?

Reply:
@romayalon I send the request from JS, to make sure no headers.
```
        const res = await http_utils.make_https_request({
            hostname: '{bucket-name}.s3.amazonaws.com',
            method: 'GET'
        });
```
Not 100% sure this will send request without any default header, since we cant check in AWS S3 side. It dont have any access credentials.

Reply:
@naveenpaul1 we print headers on our logs, try this code directed to NooBaa and check the logs

Reply:
@romayalon tried the same even this request have header info
`Anonymous request: GET / { host: 'localhost:6443', connection: 'close' }`

Reply:
@naveenpaul1 do you have an anonymous user in your system?

Reply:
ok so create one and compare the behavior
when I compare on AWS these 2 - 
1. curl https://{bucket-name}.s3.amazonaws.com --http1.0 -H 'Host:'
2. curl https://{bucket-name}.s3.amazonaws.com 
I see that only the second one returns a response as you mentioned, I expect NooBaa to have similar behavior

Reply:
Why do you run the curl with `--http1.0` in the first option?

Reply:
I'm not sure it is related to the PR, but for the manual tests - I see in AWS `AccessDenied`, where did I wrong? (I tried to follow the testing steps)

(no missing HOST header)
1) I tried to run: `curl https://{bucket-name}.s3.amazonaws.com` with my bucket:
- `curl https://shira-bucket.s3.amazonaws.com`
- and also `AWS_ACCESS_KEY_ID="" AWS_SECRET_ACCESS_KEY="" curl https://shira-bucket.s3.amazonaws.com`

and in both cases I saw `AccessDenied`.

2) When I tried: `curl https://localhost:6443 --http1.0 --insecure` (after running `sudo node src/cmd/nsfs --debug 5`)
- before I had anonymous account it was `InvalidAccessKeyId`
- after I created anonymous account (`sudo node src/cmd/manage_nsfs account add --anonymous --uid 3333 --gid 3333`) I saw `ListAllMyBucketsResult`.

Reply:
@shirady I used a public bucket like noobaa-core-rpms, you get access denied because shira-bucket isn't public

Reply:
Thanks, so in this case (`curl https://noobaa-core-rpms.s3.amazonaws.com`) I can also see the `/ListBucketResult`, so it is fine (both servers return the `ListAllMyBucketsResult`).

Reply:
@shirady and you need to add the header flag with empty host
` -H 'Host:' `